webcomponents/tool/src/main/frontend/packages/sakai-grader/src/SakaiGrader.js

```diff
@@ -65,6 +65,37 @@ export class SakaiGrader extends graderRenderingMixin(gradableDataMixin(SakaiEle
     if (typeof MathJax !== "undefined") {
       MathJax.Hub.Queue([ "Typeset", MathJax.Hub ]);
     }
+
+    window.addEventListener(
+      "message",
+      e => {
+        let message = e.data;
+        if ( typeof message == "string" ) message = JSON.parse(e.data);
+        if ( message.subject == "lti.gradeChangeNotify" ) {
+          console.log("The LTI Tool changed a grade - retrieving new grade");
+          console.log(this._submission);
+          const formData = new FormData();
+          formData.valid = true;
+          formData.set("studentId", this._submission.firstSubmitterId);
+          formData.set("courseId", portal.siteId);
+          formData.set("gradableId", this.gradableId);
+          formData.set("submissionId", this._submission.id);
+          console.log(formData);
+          fetch("/direct/assignment/getGrade.json", {
+            method: "POST",
+            cache: "no-cache",
+            credentials: "same-origin",
+            body: formData
+          })
+          .then(r => r.json()).then(data => {
```

Comment:
You need the r.ok and error throwing type stuff in here, so we catch network or server errors.

Reply:
Added